doc/topics/blocking-vs-non-blocking.md

```diff
@@ -0,0 +1,140 @@
+# Overview of Blocking vs Non-Blocking
+
+This overview covers the difference between blocking and non-blocking calls in
+Node.js. This overview will refer to the event loop and libuv but no prior
+knowledge of those topics is required. Readers are assumed to have a basic
+understanding of the JavaScript language and Node.js callback pattern.
+
+> In this overview, "I/O" refers primarily to interaction with the system's disk
+and network supported by [libuv](http://libuv.org/).
+
+
+## Blocking
+
+Blocking is when the execution of additional JavaScript in the Node.js process
+must wait until an I/O operation completes. Blocking may occur when using any of
```

Comment:
Leaving the I/O in there is fine since that's the *most likely* case. It may just be worthwhile indicating that non-I/O operations can cause it to bog down as well. crypto operations, for instance.